src/Expression.h

```diff
@@ -128,4 +128,29 @@ class ParanthesisExpression : public Expression{
     {
         return "("+expression->dotPrint()+")";
     }
+};
+
+class ArrayExpression : public Expression
+{
+public:
+    std::string id;
+    std::shared_ptr<Expression> indexExpression;
+
+    ArrayExpression(std::string id, std::shared_ptr<Expression> index): id(id), indexExpression(index){}
+
+    bool evaluate(std::set<std::string> &state)
+    {
+        return indexExpression->evaluate(state);
```

Comment:
seems weird that we assume that the index could not result in potential dangerous behavior, if the tainted value access memory outside the bound of the array it would be uncertain what value x would be. so i believable its better to overapproximate.
However there is still needed to check if any of the indexes of the array is tainted as well.

Reply:
Lets discuss this with the supervisor